sympy/integrals/tests/test_risch.py

```diff
@@ -636,6 +637,19 @@ def test_risch_integrate():
 
     assert risch_integrate(0, x) == 0
 
+    # also tests prde_cancel()
+    e1 = log(x/exp(x) + 1)
+    ans1 = risch_integrate(e1, x)
+    assert ans1 == (x*log(x*exp(-x) + 1) + NonElementaryIntegral((x**2 - x)/(x + exp(x)), x))
+    assert cancel(diff(ans1, x) - e1) == 0
```

Comment:
I've used `cancel` instead of `simplify`, as `cancel` is more specific (also taking a little less time). Also the end of 2nd para of https://github.com/sympy/sympy/issues/5109#issue-28950562 .